src/generic/grid.cpp

```diff
@@ -6935,6 +7149,116 @@ void wxGrid::DrawRowLabel( wxDC& dc, int row )
                    rect, hAlign, vAlign, wxHORIZONTAL);
 }
 
+void wxGrid::DrawSelection(wxDC& paintDC, wxGridWindow *gridWindow, const wxRect& renderExtent)
+{
+    if ( !IsUsingTransparentSelection() || !IsSelection() )
+        return;
+
+    const auto& polygons = m_selection->GetPolyPolygon(renderExtent);
+
+    wxRect updateRect = polygons.GetBoundingBox();
+
+    if ( updateRect.IsEmpty() )
+    {
+        return; // nothing to draw
+    }
+
+#if wxUSE_GRAPHICS_CONTEXT && !(defined(__WXOSX__) || defined(__WXGTK3__) || defined(__WXQT__))
+    wxGCDC gdc;
+    auto renderer = wxGraphicsRenderer::GetDefaultRenderer();
+    auto context  = renderer->CreateContext(static_cast<wxPaintDC&>(paintDC));
+
+    gdc.SetBackground(GetBackgroundColour());
+    gdc.SetGraphicsContext(context);
+
+    wxDC& dc = gdc;
+#else
+    wxDC& dc = paintDC;
+#endif
+
+    const wxPoint oldOrig = paintDC.GetLogicalOrigin();
+    const wxPoint orig    = paintDC.GetDeviceOrigin();
+    const wxPoint offset  = GetGridWindowOffset(gridWindow);
+
+    wxRect clipRect;
+
+    if ( renderExtent.IsEmpty() )
+    {
+        dc.SetLogicalOrigin(orig.x + offset.x, orig.y + offset.y);
+        clipRect = gridWindow->GetClientRect();
+    }
+    else // We are rendering on external DC
+    {
+        clipRect = renderExtent;
+    }
+
+    wxON_BLOCK_EXIT_OBJ2(dc, wxDC::SetLogicalOrigin, oldOrig.x, oldOrig.y);
+
+    clipRect.Offset(offset);
+
+    // Don't draw selection edges on frozen borders.
+    if ( offset.x > 0 )
+        clipRect.x += 1;
+    if ( offset.y > 0 )
+        clipRect.y += 1;
+
+    wxDCClipper clip(dc, clipRect);
+
+    const wxColour colBg = GetSelectionBackground();
+
+#ifdef __WXMSW__
```

Comment:
I'd appreciate a comment explaining why do we do this only under MSW as it doesn't seem obvious.

Reply:
I wanted to make the selection outline more bold but after thinking more about it I think I'll revert this as it doesn't seem necessary.

Here is how it looks without this dirty hack:

In light mode:
![grid_light](https://github.com/wxWidgets/wxWidgets/assets/7704771/12f09e8e-baed-4f1f-81be-a12d9d1c2b70)

In dark mode:
![grid_dark](https://github.com/wxWidgets/wxWidgets/assets/7704771/d283ae7c-5efd-43d1-a629-000c347c04f6)